### PR TITLE
Add an 'unparse' function for string search queries

### PR DESCRIPTION
**Pull request by nickstenning**
This commit adds a function which can be used to transform a dict or MultiDict object representing a search query back into a string query. This will be useful for activity pages because it allows us to manipulate the user's query (by, for example, removing terms) and then turn it back into a string where it can be used in a URL.

### CODE REVIEW

File: src/memex/search/parser.py [hunk 1]
```diff
@@ -76,6 +76,31 @@ def parse(q):
...
+
+    This can be considered the reverse of the
+    :py:func:`memex.search.parser.parse` function, as it can be used to
+    transform the MultiDict returned from that function back into a string
+    query.
+    """
+    terms = []
+
+    try:
+        items = q.iteritems(multi=True)
```
Comment (robertknight) [inline src/memex/search/parser.py]: The `webob.multidict.MultiDict` object returned by `memex.search.parser.parse` does not support a 'multi' keyword argument. It looks like that is a feature of `werkzeug.datastructures.MultiDict`. Is there any reason that we would be using that type with this function?

**Comment by robertknight** [comment]
Aside from the query about the `multi` argument to `iteritems` this LGTM.

File: src/memex/search/parser.py (hunk 1 above)
Comment (nickstenning) [inline src/memex/search/parser.py]: That is a *very* good point. Good spot. Given all the tests are passing I guess that `iteritems` from `webob.multidict.MultiDict` does "the right thing". I'll check.

Comment (nickstenning) [inline src/memex/search/parser.py]: Fixed.

**Pull request merged**